web/src/invite.ts

```diff
@@ -475,10 +443,48 @@ function open_invite_user_modal(e: JQuery.ClickEvent<Document, undefined>): void
         };
 
         $("#invite-user-form .setup-tips-container").html(render_invite_tips_banner(context));
+
+        const toggler = components.toggle({
+            html_class: "invite_users_option_tabs large allow-overflow",
+            selected: 0,
+            child_wants_focus: true,
+            values: [
+                {label: $t({defaultMessage: "Send invite email"}), key: "invite-email-tab"},
+                {label: $t({defaultMessage: "Create invite link"}), key: "invite-link-tab"},
+            ],
+            callback(_name, key) {
+                switch (key) {
+                    case "invite-email-tab":
+                        $("#invitee_emails_container").show();
+                        break;
+                    case "invite-link-tab":
+                        $("#invitee_emails_container").hide();
+                        break;
+                }
+                toggle_invite_submit_button();
+                reset_error_messages();
+            },
+        });
+        const $container = $("#invite_users_option_tabs_container");
+        if (!settings_data.user_can_invite_users_by_email()) {
+            toggler.disable_tab("invite-email-tab");
+            toggler.goto("invite-link-tab");
+        }
+        if (!settings_data.user_can_create_multiuse_invite()) {
+            toggler.disable_tab("invite-link-tab");
+        }
+        const $elem = toggler.get();
+        $container.append($elem);
+        setTimeout(() => {
+            $(".ind-tab.selected").trigger("focus");
+            toggle_invite_submit_button();
```

Comment:
Is it necessary to call `toggle_invite_submit_button` inside `setTimeout`.

Reply:
After the changes you did here, I am getting an error while opening the modal.

I dn't guess you need to call `toggle_invite_submit_button` here as this will only be called initially and rest of the work for keeping the button state synced with selected tab will be done in `callback` function.

We already make the button disabled at the top by just using jquery `prop` method and then if the user is not allowed to send email invites then we call the `toggle.goto` to change tab to multiuse invites which again handles the state of button. So, handling the initial state of button on opening the modal is already handled.

Reply:
I've made some amendments and things are working fine now. The need to sync the state initially arose from a minor bug in the `toggle_invite_submit_button`, but that is no longer the case.